### ISSUE TITLE
Wrong cloudflared binary architecture on Apple Silicon

### ISSUE THREAD

**Issue by ghost** [HttpRafa/Modflared#40]
**Describe the bug**
Instead of cloudflared-darwin-arm64, modflared install the cloudflared-darwin-amd64 ones.

**Which version of the mod/game are you using?**
Minecraft 1.20.1/Fabric 0.15.11/Modflared 1.2.0+release.61

**To Reproduce**
1. Launch
2. Go into multiplayer
3. Add some server
4. Check log

**Expected behavior**
It should get cloudflared-darwin-arm64.
I knew the cloudflared release does not have it, but it can get from homebrew ones. (ask user to install it)

**Log**
```
[18:33:54] [pool-7-thread-2/ERROR]: Failed to start cloudflared
java.io.IOException: Cannot run program "./cloudflared-darwin-amd64" (in directory "<REDUCTED>/.minecraft/modflared/bin"): error=86, Bad CPU type in executable
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1143) ~[?:?]
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1073) ~[?:?]
	at de.rafael.modflared.tunnel.RunningTunnel.lambda$createTunnel$0(RunningTunnel.java:40) ~[modflared-1.2.0+release.61.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:840) ~[?:?]
Caused by: java.io.IOException: error=86, Bad CPU type in executable
	at java.lang.ProcessImpl.forkAndExec(Native Method) ~[?:?]
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:314) ~[?:?]
	at java.lang.ProcessImpl.start(ProcessImpl.java:244) ~[?:?]
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1110) ~[?:?]
	... 5 more
```

**Comment by ghost** [comment]
Btw, symlink cloudflared-darwin-amd64 to /opt/homebrew/bin/cloudflared works fine.

**Comment by HttpRafa** [comment]
Cloudflared doesn't seem to have an arm64 version on github.

**Comment by ghost** [comment]
Ah, I see it. Can we use binaries in $PATH instead? (as fallback or preferred so we don't have to download/update it).

**Comment by HttpRafa** [comment]
It would be possible to check whether cloudflared is already in the PATH and then simply use that version without having to install Cloudflared separately.

**Comment by HttpRafa** [comment]
I have to backport the changes to 1.20.1 but it should now work in 1.21. Closed by https://github.com/HttpRafa/modflared/commit/bc9198fcc483ca52e7b16d56a4b635763033d78d. If you find any issue feel free to reopen this issue

**Comment by MrWillCom** [comment]
I enncountered this issue just now. Cloudflare provides the binary for macOS arm64 nowadays. So I replaced `cloudflared-darwin-amd64` with the [`cloudflared` in `cloudflared-darwin-arm64.tgz`](https://github.com/cloudflare/cloudflared/releases/tag/2026.2.0) and everything works correctly. Maybe it is possible to fix this issue?

Sometimes modflared needs `cloudflared-darwin-arm64` as well to work. (though after replacement, two files are the same)
